### PR TITLE
[IOTDB-4377] Fix TTimePartitionSlot count and DataPartition inherit policy bug

### PR DESCRIPTION
**Pull request by CRZbulabula**
I've fixed two bugs in this PR:

1. The SeriesSlot count in the result of show regions are incorrect. 
2. The DataPartitions with larger TTimePartitionSlot can't inherit their predecessor when a batch of contiguous DataPartition are demanded to create synchronously.

I solved these two bugs with one PR because it's too difficult to write stronger test for me if I break them apart.

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/StorageGroupPartitionTable.java [hunk 1]
```diff
@@ -207,17 +205,14 @@ public TConsensusGroupId getPrecededDataPartition(
...
-    AtomicInteger total = new AtomicInteger(0);
-    deltaMap.forEach(
-        ((consensusGroupId, delta) -> {
-          total.getAndAdd(delta.get());
-          regionGroupMap.get(consensusGroupId).addCounter(delta.get());
-        }));
-    seriesPartitionSlotsCount.getAndAdd(total.get());
+    // Update counter
+    groupDeltaMap.forEach(
+        ((consensusGroupId, deltaMap) ->
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/StorageGroupPartitionTable.java:214]: deltaMap, Map<TSeriesPartitionSlot, AtomicLong>, AtomicLong value is always 0, do we need execute the `AtomicLong` method?

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/RegionGroup.java [hunk 2]
```diff
@@ -28,27 +29,34 @@
...
   private final TRegionReplicaSet replicaSet;
 
+  // Map<TSeriesPartitionSlot, TTimePartitionSlot Count>
+  // For SchemaRegion, each SeriesSlot constitute a SchemaPartition.
   // For DataRegion, a SeriesSlot and a TimeSlot constitute a DataPartition.
   // Eg: A DataRegion contains SeriesSlot-1 which has TimeSlot-1, TimeSlot-2 and Timeslot-3,
   // then (SeriesSlot-1 -> TimeSlot-1) constitute a DataPartition.
-  // For SchemaRegion, each SeriesSlot constitute a SchemaPartition.
-  private final AtomicLong slotCount;
+  private final Map<TSeriesPartitionSlot, AtomicLong> slotCountMap;
```
Comment (Beyyes) [inline confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/RegionGroup.java:46]: Maybe the value of Map<TSeriesPartitionSlot, AtomicLong> is not used?

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java [hunk 3]
```diff
@@ -441,7 +442,7 @@ public DataSet getSchemaNodeManagementPartition(List<String> matchedStorageGroup
   /** Get region information */
   public DataSet getRegionInfoList(GetRegionInfoListPlan regionsInfoPlan) {
     RegionInfoListResp regionResp = new RegionInfoListResp();
-    List<TRegionInfo> regionInfoList = new ArrayList<>();
+    List<TRegionInfo> regionInfoList = new Vector<>();
```
[the remark this answers is not in the document]
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/PartitionInfo.java:445]: Vector supports concurrency safety while ArrayList doesn't

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/StorageGroupPartitionTable.java (hunk 1 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/StorageGroupPartitionTable.java:214]: For SchameRegionGroup, the AtomicLong value is always 0. For DataRegionGroup, the AtomicLong value represents the TimeSlotCount of the specified TSeriesPartitionSlot. Currently I mingle the SchemaRegionGroup and the DataRegionGroup into the RegionGroup, for which might cause some misunderstands. I'll optimize these code in the future~

File: confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/RegionGroup.java (hunk 2 above)
Comment (CRZbulabula) [reply confignode/src/main/java/org/apache/iotdb/confignode/persistence/partition/RegionGroup.java:46]: The value of Map<TSeriesPartitionSlot, AtomicLong> will be used to show the TimeSlotCount of show regions.

**Review by Beyyes — APPROVED** [review]
LGTM

**Pull request merged**